main_dialog.py

```diff
@@ -0,0 +1,308 @@
+        self.push_button_5.clicked.connect(self.clear_all_items)
+        self.show()
+
+        self.list_widget_1.setAcceptDrops(True)
+        self.list_widget_1.setMouseTracking(True)
+        self.list_widget_1.verticalScrollBar()
+        self.list_widget_1.horizontalScrollBar()
+        self.list_widget_1.setDragDropMode(QAbstractItemView.InternalMove)
+        self.list_widget_1.setSelectionMode(
+             QAbstractItemView.ExtendedSelection)
+
+        self.list_widget_1.currentItemChanged.connect(self.items_changed)
+
+    # Actions for mouse right click #
+        quit_action_1 = QAction("Zamknij", self, shortcut="Ctrl+Q",
+                                triggered=QApplication.instance().quit)
+        quit_action_2 = QAction("Czyść", self, triggered=self.clear_all_items)
+        quit_action_3 = QAction("Zaznacz wszystko", self, shortcut="Ctrl+A",
+                                triggered=self.select_items)
+        quit_action_4 = QAction("Usuń zaznaczone", self, shortcut="Del",
+                                triggered=self.clear_selected_items)
+        self.list_widget_1.addAction(quit_action_4)
+        self.list_widget_1.addAction(quit_action_3)
+        self.list_widget_1.addAction(quit_action_2)
+        self.list_widget_1.addAction(quit_action_1)
+
+    # Info for users how to add files
+        self.list_widget_1.setContextMenuPolicy(Qt.ActionsContextMenu)
+
+        self.push_button_1.setToolTip("Kliknij, aby dodać pliki do listy")
+        self.push_button_2.setToolTip("Kliknij, aby wybrać przepis")
+        self.push_button_3.setToolTip(
+            "Kliknij, aby przejść do generowania dokumentu")
+        self.push_button_4.setToolTip(
+            "Kliknij, aby usunać zaznaczone elementy")
+
+        self.push_button_5.setToolTip("Kliknij, aby wyczyścić liste")
+        settings.localPath
+        settings.zipsFolder
+
+        self.returnedFiles = []
+        self.boxIsChecked = False
+        self.selectedRecipe = str(settings.localPath + settings.zipsFolder +
+                                  settings.sets['user']['default-recipe'])
+        self.items_changed()
+
+
+# << END of: Custom Main Widget >> #
+    def aboutGPadnoc(self):
+        dialog = AboutDialog()
+        dialog.exec_()
+
+    def instruction(self):
+        dialog = InstructionDialog()
+        dialog.exec_()
+
+    def settings(self, loadedRecipe):
+        dialog = SettingsDialog(loadedRecipe)
+        self.setFont(QFont(settings.sets['user']['font-name'],
+                           int(settings.sets['user']['font-size'])))
+        dialog.exec_()
+
+
+# << Listwidget handling >> #
+
+    # clear current selected item
+    def clear_selected_items(self):
+        for select in self.list_widget_1.selectedItems():
+            self.list_widget_1.takeItem(self.list_widget_1.row(select))
+        self.items_changed()
+
+    def check_extensions(self):
+        if (self.list_widget_1.count()):
```

Comment:
Dziwna konwencja, żeby w Pythonie wyrażenia w ifie brać w nawiasy - raczej jest to zbędne